include/execution.hpp

```diff
@@ -2252,142 +2183,28 @@ namespace std::execution {
-                  __bind_back<__defer<__error_types_of_t>, _Env, __q<__types>>,
-                  __bind_front<
-                    __concat<__munique<__q<_Variant>>>,
-                    __types<exception_ptr>>>>;
-
-          static constexpr bool sends_stopped =
-            __result_senders_t<
-              __transform<
-                __bind_back_q1<__sends_stopped, _Env>,
-                __right_fold<
-                  __sends_stopped<_Sender, _Env>,
-                  __q2<__or>>>>::value;
+      template <class _Env, class _Fun, class _Set, class _Ret, class... _Args>
+          requires (!same_as<_Set, _Ret>)
+        struct __tfx_signal<_Env, _Fun, _Set, _Ret(_Args...)> {
+          using type = completion_signatures<_Ret(_Args...)>;
         };
 
-      template <class _Sender, class _Env, class _Fun>
-          requires sender<_Sender, _Env> &&
-            __valid<__stopped_senders_of, _Sender, _Env, _Fun, __typed_senders<_Env>>
-        struct __completion_sigs<_Sender, _Env, _Fun, set_stopped_t>
-        {
-          template <class _Continuation>
-            using __result_senders_t =
-              __stopped_senders_of<_Sender, _Env, _Fun, _Continuation>;
-
-          template <template <class...> class _Tuple, template <class...> class _Variant>
-            using value_types =
-              __result_senders_t<
-                __transform<
-                  __bind_back<__defer<__value_types_of_t>, _Env, __q<_Tuple>, __q<__types>>,
-                  __bind_front<
-                    __concat<__munique<__q<_Variant>>>,
-                    value_types_of_t<_Sender, _Env, _Tuple, __types>>>>;
-
-          template <template <class...> class _Variant>
-            using error_types =
-              __result_senders_t<
-                __transform<
-                  __bind_back<__defer<__error_types_of_t>, _Env, __q<__types>>,
-                  __bind_front<
-                    __concat<__munique<__q<_Variant>>>,
-                    __types<exception_ptr>,
-                    error_types_of_t<_Sender, _Env, __types>>>>;
-
-          static constexpr bool sends_stopped =
-            __result_senders_t<
-              __transform<
-                __bind_back_q1<__sends_stopped, _Env>,
-                __right_fold<false_type, __q2<__or>>>>::value;
+      template <class _Env, class _Fun, class _Set, class... _Args>
+          requires invocable<_Fun, _Args...> &&
+            sender<invoke_result_t<_Fun, _Args...>, _Env> &&
+            __is_typed_completions<
+              completion_signatures_of_t<invoke_result_t<_Fun, _Args...>, _Env>>
+        struct __tfx_signal<_Env, _Fun, _Set, _Set(_Args...)> {
+          using __completions =
+            completion_signatures_of_t<invoke_result_t<_Fun, _Args...>, _Env>;
+          using type =
+            __minvoke2<
+              __push_back<__q<completion_signatures>>,
+              __completions,
+              set_error_t(exception_ptr)>;
```

Comment:
> Does these mean let_stopped (I think that's where we are?) requires exception_ptr?

I'm being sloppy here. What this code _should_ be doing is checking to see whether the function call is potentially throwing. If so, it should add `set_error_t(exception_ptr)` to the list of completions. Otherwise, it should leave it off.

> New Line 2229, there's comments about "for the purposes of the receiver concept..."
Is that still accurate?

Nope, no longer accurate. I'll remove it.